lib/saluki-core/src/topology/blueprint.rs

```diff
@@ -63,78 +61,84 @@ impl TopologyBlueprint {
+            interconnect_capacity: DEFAULT_INTERCONNECT_CAPACITY,
+        }
     }
 
-    fn update_bounds_for_interconnect(&mut self) {
-        // Increment the size of our interconnects-related bounds to account for a newly-added transform or destination.
-        self.component_registry
-            .get_or_create("interconnects")
-            .bounds_builder()
-            .minimum()
-            .with_array::<FixedSizeEventBuffer>("fixed size event buffers", 128);
+    /// Sets the size of event buffers in the global event buffer pool.
+    ///
+    /// The global event buffer pool is used by components to acquire an event buffer that can be used to forward events
+    /// to the next component in the topology. Each individual event buffer will be allocated to hold `buffer_size` events.
+    ///
+    /// Defaults to 1024 events.
+    pub fn with_global_event_buffer_pool_size(&mut self, buffer_size: NonZeroUsize) -> &mut Self {
+        self.event_buffer_pool_buffer_size = buffer_size;
+        self.recalculate_bounds();
+        self
     }
 
-    /// Sets the global event buffer pool sizing.
+    /// Sets the component interconnect capacity.
     ///
-    /// The global event buffer pool is used by components to acquire an event buffer that can be used to forward events
-    /// to the next component in the topology.
+    /// This controls how many outstanding event buffers can be queued for each component before backpressure is applied
+    /// to upstream components. Changing this value allows trading off between the peak memory usage when processing a high
+    /// volume of events and the amount of buffering that takes place to smooth out processing speed and efficiency.
     ///
-    /// Each individual event buffer will be allocated to hold `buffer_size` events, and the pool will be sized to hold
-    /// a minimum of `size_min` buffer, and up to a maximum of `size_max` event buffers.
-    pub fn with_global_event_buffer_pool_size(
-        &mut self, buffer_size: NonZeroUsize, size_min: NonZeroUsize, size_max: NonZeroUsize,
-    ) -> &mut Self {
-        self.event_buffer_pool_buffer_size = buffer_size.get();
-        self.event_buffer_pool_size_min = size_min.get();
-        self.event_buffer_pool_size_max = size_max.get();
-
-        // Account for our global event buffer pool, which has a lower and upper bound for the pooled object limit, but
-        // using fixed-size event buffers.
+    /// Defaults to 128.
+    pub fn with_component_interconnect_capacity(&mut self, capacity: NonZeroUsize) -> &mut Self {
+        self.interconnect_capacity = capacity;
+        self.recalculate_bounds();
+        self
+    }
+
+    fn recalculate_bounds(&mut self) {
+        // Adjust the bounds related to interconnects.
+        //
+        // Every non-source component has an interconnect.
+        let total_interconnect_capacity =
+            (self.transforms.len() + self.destinations.len()) * self.interconnect_capacity.get();
```

Comment:
We don't do it now, but technically you could have more edges in the graph than this (e.g. two parallel transforms both sending to two sinks).

Reply:
Hmm... I don't think that changes anything?

Each non-source component has a single interconnect channel, regardless of how many upstream components are connected to it.

Reply:
😂 No worries.